src/readers/src/virtex_data_classes.cpp

```diff
@@ -0,0 +1,107 @@
+#include "virtex_data_classes.h"
+
+#include <databases.h>
+#include <shapes.h>
+
+#include <ranges>
+
+namespace {
+size_t getNewId() {
+	static size_t id{0};
+	++id;
+	return id;
+}
+
+std::pair<float, float> getSphereProjection(float center, size_t offset,
+                                            float radius) {
+	return {center + offset - radius, center + offset + radius};
+}
+}  // namespace
+
+namespace Readers {
+LayerType BaseLayer::getType() { return layer_type_; };
```

Comment:
1) by our codestyle primitive getter should be called in the same way as valiable exept "_" in the end, so for current case it will be layer_type()
2) getter name and field has different name. It feels like "layer" can be omitted it the fileld name, so final result should be 
LayerType BaseLayer::type() { return type_; };